### ISSUE TITLE
Build breakage (missing C99 standard) after 45222ad76c33f49462e70916353bda3f1c2d9f71

### ISSUE THREAD

**Issue by rainerjung** [jvoisin/snuffleupagus#410]
Commit 45222ad76c33f49462e70916353bda3f1c2d9f71 breaks snuffleupagus build on Platforms, where C99 is not the default standard. Before the commit, configure added "-std=gnu99" to CFLAGS. After the commit, no standard is chosen.

Plarforms like SuSE Linux Enterprise Server 11 and 12 and RedHat Enterprise Linux 6 and 7 support C99 with the "-std=..." option, but do not use it by default. Newer GCC like the ones bundled with SLES 15 and RHEL 8 use C99 or an even newer standard by default, so the problem does not occur there.

Compilation fails with errors, eg. because "restrict" is only a C99 keyword.

Please consider adding back AC_PROG_CC_STDC (or instead AC_PROG_CC_C99).

Thank you.

**Comment by jvoisin** [comment]
Apparently, `AC_PROG_CC_STDC` is deprecated:

```
configure.ac:22: the top level
config.m4:24: warning: The macro `AC_PROG_CC_STDC' is obsolete.
config.m4:24: You should run autoupdate.
./lib/autoconf/c.m4:1666: AC_PROG_CC_STDC is expanded from...
config.m4:24: the top level
```

**Comment by jvoisin** [comment]
It should be fixed by 192a24a035d5ef887e4163925156502552b78c7d

Can you confirm?

**Comment by rainerjung** [comment]
Thanks for the change. Unfortunateyit does not work, here's the reason:

When building snuffleupagus from git, I need to run phpize to eg. create configure. phpize uses autoconf on the target system. On these older Linux releases autoconf is also slightly older, eg. version 2.69. That version of autoconf not only supports AC_PROG_CC_STDC, it also needs it, because it had not already been merged into AC_PROG_CC like it was done with autoconf 2.70.

Most projects do not need to care, because autoconf only ever runs on machines doing release preparations and then the final configure script is shipped. To support older systems even for the autoconf creation, I found the following workaround in another project:

AC_PROG_CC
# AC_PROG_CC_STDC was merged in AC_PROG_CC in autoconf 2.70
m4_version_prereq([2.70], [], [AC_PROG_CC_STDC])

I will test this later today and then report back, how it works for me on an old and a new system.

Thanks and regards,

Rainer

**Comment by rainerjung** [comment]
I can confirm, that adding

```
# AC_PROG_CC_STDC was merged in AC_PROG_CC in autoconf 2.70
m4_version_prereq([2.70], [], [AC_PROG_CC_STDC])
```

directly after `AC_PROG_CC()` works for me on old and new systems. You might want to drop or adjust the comment line according to your stylistic wishes.

Thanks and regards,

Rainer

**Comment by jvoisin** [comment]
Thank you very much for investigating!

I pushed 5d1d7365d981f260023a193c5738413c4bfaa6bb
